### PR TITLE
Fix deadlock in Server Close()

### PR DESCRIPTION
**Pull request by makon**
Mainly this PR fixes the deadlock in Close().
It also has some minor fixes for the Readme and the unit tests.

I added a travis ci config file but unfortunatelly I can't configure it because I am not the owner of the repo. I would really appreciate it when you could set it up. It should be really easy. 
https://docs.travis-ci.com/user/tutorial/
https://docs.travis-ci.com/user/languages/go/

### CODE REVIEW

**Comment by zonque** [comment]
Thanks for your contribution!

Unfortunately, this branch has outdated now, sorry for that. Could you rebase please?

File: server_test.go [hunk 1]
```diff
@@ -79,6 +81,30 @@ func TestBasic(t *testing.T) {
 		t.Fatal("Entry group must initially be empty")
 	}
 
-	egc.Free()
+	egc.free()
+
```
Comment (zonque) [inline server_test.go:85]: Stray newline?

File: server.go [hunk 2]
```diff
@@ -136,7 +136,7 @@ func ServerNew(conn *dbus.Conn) (*Server, error) {
 }
 
 func (c *Server) Close() {
-	<-c.quitChannel
+	c.quitChannel <- true
```
Comment (zonque) [inline server.go:139]: This could also become a `chan struct{}` I guess.

File: server_test.go (hunk 1 above)
Comment (zonque) [reply server_test.go:85]: `go fmt` should also catch this :)

**Review by zonque — CHANGES_REQUESTED** [review]
Please rebase and fix the style things, then we can quickly merge. Sorry for the delay on this, I was travelling for a longer while.

**Comment by zonque** [comment]
Would you like to rebase this PR? I'd like like to merge this ...

**Comment by makon** [comment]
Sorry for the late response, I will redo it on a new fork

**Pull request closed without merging**